atom/browser/api/atom_api_notification.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2014 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ATOM_BROWSER_API_ATOM_API_NOTIFICATION_H_
+#define ATOM_BROWSER_API_ATOM_API_NOTIFICATION_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "atom/browser/api/trackable_object.h"
+#include "atom/browser/ui/notification_observer.h"
+#include "base/strings/utf_string_conversions.h"
+#include "brightray/browser/notification.h"
+#include "brightray/browser/notification_presenter.h"
+#include "native_mate/handle.h"
+#include "ui/gfx/image/image.h"
+
+namespace atom {
+
+namespace api {
+
+class Notification : public mate::TrackableObject<Notification>,
+                     public NotifictionObserver {
```

Comment:
There is no need to add `NotifictionObserver` and `AtomNotificationDelegateAdapter` in this case, `api:: Notification` can just inherit `brightray::NotificationDelegate`:

```c++
class Notification : public mate::TrackableObject<Notification>,
                     public brightray::NotificationDelegate {
  void NotificationDisplayed() override;
};

Notification::NotificationDisplayed() {
  Emit("show");
}

void Notification::Show() {
  notification_ = presenter_->CreateNotification(this);
}
```

We also need to guard the case when `api::Notification` is destroyed before `brightray::Notification`, it would require a few changes to `brightray::Notification`:

```c++
namespace api {

Notification::~Notification() {
  notification_->set_delegate(nullptr);
}

}
```

```c++
namespace brightray {

Notification::NotificationDisplayed() {
  if (delegate())
    delegate()->NotificationDisplayed();
}

}